docs/src/reconciling-local-changes.md

```diff
@@ -0,0 +1,234 @@
+# Reconciling changes made outside of TPA
+Any changes made to a TPA created cluster that are not performed by
+changing the TPA configuration will not be saved in `config.yml`. This
+means that your cluster will have changes that the TPA configuration
+won't be able to recreate.
+
+This page shows how configuration is managed with TPA and the preferred
+ways to make configuration changes. We then look at strategies to make,
+and reconcile, the results of making manual changes to the cluster.
+
+## Why might I need to make manual configuration changes?
+The most common scenario in which you may need to make configuration
+changes outside of TPA is if the operation you are performing is not
+supported by TPA. The two most common such operations are destructive
+changes, such as removing a node, and upgrading the major version of
+Postgres.
+
+### Destructive changes
+In general TPA will not remove previously deployed elements of a
+cluster, even if these are removed from `config.yml`. This sometimes
+surprises people because a strictly declarative system should always
+mutate the deployed artifacts to match the declaration. However, making
+destructive changes to production database can have serious consequences
+so it is something we have chosen not to support.
+
+### Major-version Postgres upgrades
+TPA does not yet provide an automated mechanism for performing major
+version upgrades of Postgres. Therefore if you need to perform an
+in-place upgrade on an existing cluster this must be performed using
+other tools such as pg_upgrade or [bdr_pg_upgrade](https://www.enterprisedb.com/docs/pgd/latest/upgrades/bdr_pg_upgrade/#bdr_pg_upgrade-command-line).
+
+## What can happen if changes are not reconciled?
+A general issue with unreconciled changes is that if you deploy a new
+cluster using your existing `config.yml`, or provide your `config.yml` to
+EDB Support in order to reproduce a problem, it will not match the
+original cluster. In addition, there is potential for operational
+problems should you wish to use TPA to manage that cluster in future.
+
+The operational impact of unreconciled changes varies depending on the
+nature of the changes. In particular whether the change is destructive,
+and whether the change blocks TPA from running by causing an error or
+invalidating the data in `config.yml`.
+
+### Non-destructive, non-blocking changes
+Additive changes are often accommodated with no immediate operational
+issues, Consider manually adding an extension. Because TPA does not make
+destructive changes, the extension will not be removed when `tpaexec
+deploy` is next run.
+
+It should be noted however that, TPA will not make any attempt to modify
+the `config.yml` file to reflect this change and the new extension will
+be omitted from `tpaexec upgrade` which could lead to incompatible
+software versions existing on the cluster.
```

Comment:
While TPA may not remove the extension itself, some extensions require configuration changes in postgresql's configuration files. So again, when a subsequent deploy overwrites your config, you can potentially end up in an inconsistent state as far as this particular extension (or set of extensions) go. For example, you may have directly modified `shared_preload_libraries` to add an extension which will be overwritten in a subsequent deploy and potentially leave you with unexpected problems.